wordpress-mobile/WordPress-iOS
# Reader post comments: show comments past the first page

Fixes #17353

This fixes the issue where all comments were not being displayed on a post. 

The bug was a result of [the change](https://github.com/wordpress-mobile/WordPress-iOS/pull/16980) that set default values for all `Comment` attributes, including `parentID`. The fetch request predicates used for determining threaded comment pagination were not updated and were checking for `parentID` incorrectly.

To test:
- Go to the Reader.
- Select a post with a lot of comments (specifically, one with more that 20 _top level_ comments). 
  - Example: https://wordpress.com/blog/2021/07/13/blogging-reminders-push-notifications-wordpress-apps/.
- Scroll to the bottom.
- Verify more comments load.
- Compare with the web version and verify all the comments load.

The example post:

| Example Post | End of Comments Before | End of Comments After |
|--------|-------|-------|
| ![post](https://user-images.githubusercontent.com/1816888/138977454-e87e8e97-99ef-44b7-9f45-40e2332e7aa5.png) | ![before](https://user-images.githubusercontent.com/1816888/138977450-51eaead1-d619-4812-974d-deb6d2dd1f73.png) | ![after](https://user-images.githubusercontent.com/1816888/138977458-041dd8a7-d5d0-43d8-a03c-35e9f99ad98c.png) |

## Regression Notes
1. Potential unintended areas of impact
Should be none. This is the only place the modified code is used.

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Verified the correct comments are displayed.

3. What automated tests I added (or what prevented me from doing so)
N/A

PR submission checklist:
